sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/AccessPolicyAsyncTests.java

```diff
@@ -0,0 +1,1024 @@
+
+    @Test
+    public void setAccessPolicyMinAccess() {
+        setAccessPolicySleepAsync(ccAsync, PublicAccessType.CONTAINER, null);
+        StepVerifier.create(ccAsync.getProperties())
+            .assertNext(r -> assertEquals(r.getBlobPublicAccess(), PublicAccessType.CONTAINER))
+            .verifyComplete();
+    }
+
+    @Test
+    public void setAccessPolicyMinIds() {
+        BlobSignedIdentifier identifier = new BlobSignedIdentifier()
+            .setId("0000")
+            .setAccessPolicy(new BlobAccessPolicy()
+                .setStartsOn(testResourceNamer.now().atZoneSameInstant(ZoneId.of("UTC")).toOffsetDateTime())
+                .setExpiresOn(testResourceNamer.now().atZoneSameInstant(ZoneId.of("UTC")).toOffsetDateTime()
+                    .plusDays(1))
+                .setPermissions("r"));
+
+        List<BlobSignedIdentifier> ids = Collections.singletonList(identifier);
+
+        setAccessPolicySleepAsync(ccAsync, null, ids);
+
+        StepVerifier.create(ccAsync.getAccessPolicy())
+            .assertNext(r -> assertEquals(r.getIdentifiers().get(0).getId(), "0000"))
+            .verifyComplete();
+    }
+
+    @Test
+    public void setAccessPolicyError() {
+        ccAsync = primaryBlobServiceAsyncClient.getBlobContainerAsyncClient(generateContainerName());
+
+        StepVerifier.create(ccAsync.setAccessPolicy(null, null))
+            .verifyError(BlobStorageException.class);
+    }
+
+    @Test
+    public void getAccessPolicy() {
+        BlobSignedIdentifier identifier = new BlobSignedIdentifier()
+            .setId("0000")
+            .setAccessPolicy(new BlobAccessPolicy()
+                .setStartsOn(testResourceNamer.now())
+                .setExpiresOn(testResourceNamer.now().plusDays(1))
+                .setPermissions("r"));
+        List<BlobSignedIdentifier> ids = Collections.singletonList(identifier);
+        setAccessPolicySleepAsync(ccAsync, PublicAccessType.BLOB, ids);
+
+        StepVerifier.create(ccAsync.getAccessPolicyWithResponse(null))
+            .assertNext(r -> {
+                assertResponseStatusCode(r, 200);
+                assertEquals(r.getValue().getBlobAccessType(), PublicAccessType.BLOB);
+                assertTrue(validateBasicHeaders(r.getHeaders()));
+                assertEquals(r.getValue().getIdentifiers().get(0).getAccessPolicy().getExpiresOn(),
+                    identifier.getAccessPolicy().getExpiresOn());
+                assertEquals(r.getValue().getIdentifiers().get(0).getAccessPolicy().getStartsOn(),
+                    identifier.getAccessPolicy().getStartsOn());
+                assertEquals(r.getValue().getIdentifiers().get(0).getAccessPolicy().getPermissions(),
+                    identifier.getAccessPolicy().getPermissions());
+            })
+            .verifyComplete();
+    }
+
+    //block blob async
+
+    @Test
+    public void stageBlockFromUrlMin() {
+        setAccessPolicySleepAsync(ccAsync, PublicAccessType.CONTAINER, null);
```

Comment:
Is this test really a AccessPolicy test? It looks like we make the container publicly accessible but then we're using a container and blob client that uses credentials.

Appears there are a few tests like this.